From 8dac8e434be7dfb58086e04034186df34657d71f Mon Sep 17 00:00:00 2001
From: effects3d <effects3dee@gmail.com>
Date: Sun, 30 Aug 2026 15:38:27 +0200
Subject: [PATCH 13/18] wineboot: Add WINEBOOT_HIDE_DIALOG environment variable

Required for hiding Wine's boot dialog during Vinegar's preparation
process. As of 2026-09-27, this is worked around by killing the Wine
server before the boot dialog gets a chance to appear.

Source: https://gitlab.winehq.org/wine/wine/-/merge_requests/10154
Signed-off-by: effects3d <effects3dee@gmail.com>
---
 programs/wineboot/wineboot.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/programs/wineboot/wineboot.c b/programs/wineboot/wineboot.c
index 0a88a98..99163f6 100644
--- a/programs/wineboot/wineboot.c
+++ b/programs/wineboot/wineboot.c
@@ -1549,7 +1549,9 @@ static HWND show_wait_window(void)
 {
     HWND hwnd = CreateDialogParamW( GetModuleHandleW(0), MAKEINTRESOURCEW(IDD_WAITDLG), 0,
                                     wait_dlgproc, (LPARAM)prettyprint_configdir() );
-    ShowWindow( hwnd, SW_SHOWNORMAL );
+    WCHAR *hide = _wgetenv( L"WINEBOOT_HIDE_DIALOG" );
+    if (hide == NULL)
+        ShowWindow( hwnd, SW_SHOWNORMAL );
     return hwnd;
 }
 
-- 
2.56.0

